src/wp-includes/functions.php

```diff
@@ -3886,6 +3886,20 @@ function _default_wp_die_handler( $message, $title = '', $args = array() ) {
 		$message  .= "\n<p><a href='javascript:history.back()'>$back_text</a></p>";
 	}
 
+	// Ensure HTTP 500 status code for critical errors, even if headers were already sent.
+	$is_critical_error = false;
+	if ( is_wp_error( $message ) ) {
+			$error_code        = $message->get_error_code();
+			$is_critical_error = ( 'internal_server_error' === $error_code );
```

Comment:
No need to save the error code as a variable since it's only used once.

```suggestion
			$is_critical_error = ( 'internal_server_error' === $message->get_error_code() );
```

Reply:
fixed